lib/handler/redirect-handler.js

```diff
@@ -228,7 +228,7 @@ function cleanRequestHeaders (headers, removeContent, unknownOrigin) {
       }
     }
   } else if (headers && typeof headers === 'object') {
-    const entries = headers instanceof Headers ? headers.entries() : Object.entries(headers)
+    const entries = util.isIterable(headers) ? headers : Object.entries(headers)
```

Comment:
```suggestion
    const entries = typeof headers[Symbol.iterator] === 'function' ? headers : Object.entries(headers)
```

`util.isIterable` will catch `Symbol.asyncIterator` .

Reply:
yea, that does seem wrong